flask_admin/contrib/mongoengine/filters.py

```diff
@@ -221,7 +236,21 @@ def operation(self):
         return lazy_gettext('not between')
 
 
+class ReferenceFilter(BaseMongoEngineFilter):
+
+    def clean(self, value):
```

Comment:
`validate` will return `false` if `clean` raises a ValueError, but it looks like invalid ObjectIds will return this`InvalidId` exception instead: https://gist.github.com/anonymous/4fc86f0798e481095d24

You're also going to need to override the `validate` method to handle `InvalidId` exceptions.

Here's the validate method you need to override: https://github.com/flask-admin/flask-admin/blob/7445f6a486556ab1faf92fc9180af15ab24f5fd6/flask_admin/model/filters.py#L56